cli/src/complete.rs

```diff
@@ -638,6 +638,25 @@ pub fn interdiff_files(current: &std::ffi::OsStr) -> Vec<CompletionCandidate> {
     })
 }
 
+/// Specific function for completing file paths for `jj log`
+pub fn log_files(current: &std::ffi::OsStr) -> Vec<CompletionCandidate> {
+    let mut rev = parse::log_revisions()
+        // multiple -r arguments are interpreted as a union
+        .into_iter()
+        .fold("none()".into(), |mut buf: String, rev| {
+            buf.push_str("|(");
+            buf.push_str(&rev);
+            buf.push(')');
+            buf
+        });
+    if rev == "none()" {
+        rev = "@".into();
+    } else {
+        rev = format!("latest({rev})"); // limit to one
```

Comment:
It's kinda implementation detail, but `latest()` is expensive if the source set is large. We can instead use `latest(heads(..))` for now.